ujt/app.py

```diff
@@ -17,31 +17,47 @@
 from typing import Dict, List, Set, Tuple, Union, cast
 
 import dash
+import dash_bootstrap_components as dbc
 import dash_cytoscape as cyto
 import dash_html_components as html
 from dash.dependencies import Input, Output
+from dash.exceptions import PreventUpdate
 from flask_caching import Cache
 from google.protobuf.message import Message
-
-from generated.graph_structures_pb2 import (SLI, Client, Node, SLIType, Status,
-                                            UserJourney)
+from graph_structures_pb2 import (SLI, Client, Node, NodeType, SLIType, Status,
+                                  UserJourney)
 
 from . import compute_status, converters, generate_data, utils
 
+# Initialize Dash app and Flask-Cache
+cyto.load_extra_layouts()
+app = dash.Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])
+cache = Cache()
+cache.init_app(
+    app.server,
+    config={
+        "CACHE_TYPE": "filesystem",
+        "CACHE_DIR": "cache_dir"
+    },
+)
+
 
+@cache.memoize()
```

Comment:
Here is the interesting part -- memoize the call to read_local_data() so we don't actually read from disk (in the future, server) unless the cache is invalidated.